### PR TITLE
feat!: rewrite in solid.js and repos configurable in yaml and markdown

### PR DESCRIPTION
**Pull request by c0rydoras**
- full rewrite in [solid.js](https://www.solidjs.com/)
- repos are now configurable via `repos.yaml`
```yaml
repos:
  - name: Alpine
    help: |
      Add the following lines to your **/etc/apk/repositories**:

      Replace "v3.7" with your release.

      ```
      http://pkg.adfinis.com/alpine/v3.7/main
      https://pkg.adfinis.com/alpine/v3.7/main
      ```
    additional: https://mirrors.alpinelinux.org/
    official: true

  - name: Arch Linux
    help: |
      Since we are officially listed, you may just uncomment our mirror in /etc/pacman.d/mirrorlist.

      In other cases, you may add these lines to your **/etc/pacman.d/mirrorlist**:

      ```
      Server = https://pkg.adfinis.com/archlinux/$repo/os/$arch
      Server = http://pkg.adfinis.com/archlinux/$repo/os/$arch
      ```

    additional: https://www.archlinux.org/mirrors/adfinis.com/
    official: true
    localName: archlinux

 ...
```
- uses colours and font from corporate identity
- looks very similar
image
image

- should not require any changes to https://github.com/adfinis/ansible-role-repo_mirror

---
Files that need reviewing:
- https://github.com/adfinis/landingpage-repo_mirror/pull/24/files#diff-9710d6974ef8c7b745ebbce87f141d9ca79a59fbad3ed4a67ba68a9cdce32f27
- https://github.com/adfinis/landingpage-repo_mirror/pull/24/files#diff-29502bf6b85925e8b1f95138df0d8da4c370a2c7da4491dddc5eceb7e49e7776

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,37 +7,33 @@
...
 
-The page is compiled and minified using nodejs broccoli and then uploaded to our mirror.
+The page is built with vite and then uploaded to our mirror.
 
 ## Requirements
 
 NodeJS and yarn must be installed on your system.
 
-broccoli and its dependencies are installed using yarn:
+the projects dependencies are installed using yarn:
```
Comment (gianklug) [inline README.md]: The dependencies of the project

**Review by gianklug — APPROVED** [review]

File: solid/public/icons/centos-stream.png
Comment (ghost) [inline solid/public/icons/centos-stream.png:1]: what happened here? why is there text in the image?

File: repos.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
+repos:
+  - name: Alpine
+    help: |
```
Comment (ghost) [inline repos.yaml:3]: Is the "help" field optional?

File: solid/public/icons/centos-stream.png
Comment (ghost) [reply solid/public/icons/centos-stream.png:1]: Ahhh thats a symlink. Okay

File: repos.yaml (hunk 2 above)
Comment (c0rydoras) [reply repos.yaml:3]: yes

Comment (ghost) [reply repos.yaml:3]: After internal discussion, yes its optional

**Comment by ghost** [comment]
I did not review anything slightly javascript related because im too old for javascript and i have no clue about it.

**Review by hairmare — COMMENTED** [review]
not a full review yet, but what's with `solid/src/assets/solid.svg`? Is it referenced anywhere?

File: README.md [hunk 3]
```diff
@@ -7,37 +7,33 @@
...
 
-The page is compiled and minified using nodejs broccoli and then uploaded to our mirror.
+The page is built with vite and then uploaded to our mirror.
 
 ## Requirements
 
 NodeJS and yarn must be installed on your system.
 
-broccoli and its dependencies are installed using yarn:
+The dependencies of the project can be installed using yarn:
```
Comment (hairmare) [inline README.md]:
```suggestion
The dependencies of the project need to be installed using yarn:
```

Comment (hairmare) [reply README.md]: also, it does this on a fresh clone of this branch:
```console
$ yarn
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
warning solid-markdown@2.0.0: The engine "pnpm" appears to be invalid.
[3/4] Linking dependencies...
warning " > @babel/eslint-parser@7.23.3" has unmet peer dependency "@babel/core@^7.11.0".
warning "eslint-plugin-solid > @typescript-eslint/utils > @typescript-eslint/typescript-estree > ts-api-utils@1.0.3" has unmet peer dependency "typescript@>=4.2.0".
warning " > vite-plugin-fonts@0.7.0" has incorrect peer dependency "vite@^2.0.0 || ^3.0.0 || ^4.0.0".
error An unexpected error occurred: "EEXIST: file already exists, mkdir '/home/hairmare/Documents/git.repos/adsy/landingpage-repo_mirror/node_modules'".
info If you think this is a bug, please open a bug report with the information provided in "/home/hairmare/Documents/git.repos/adsy/landingpage-repo_mirror/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Review by hairmare — COMMENTED** [review]
* [x] `.travis.yml` can probs also be removed

File: README.md (hunk 3 above)
Comment (c0rydoras) [reply README.md]: should be fixed (workaround) here https://github.com/adfinis/landingpage-repo_mirror/pull/24/commits/b1213e572fbb61be58ed3d95991de80ff35d7d9c

**Comment by c0rydoras** [comment]
i forgot to remove `solid/src/assets/solid.svg`, its in the default application (`npm create vite`)

**Comment by hairmare** [comment]
you might also consider updating some of these deps before we merge:
```
Package                    Current Wanted  Latest  Package Type    URL                                                                                     
@solid-primitives/keyboard 1.2.5   1.2.6   1.2.6   dependencies    https://github.com/solidjs-community/solid-primitives/tree/main/packages/keyboard#readme
autoprefixer               10.4.16 10.4.17 10.4.17 devDependencies https://github.com/postcss/autoprefixer#readme                                          
eslint-plugin-n            16.5.0  16.6.2  16.6.2  devDependencies https://github.com/eslint-community/eslint-plugin-n#readme                              
eslint-plugin-prettier     5.1.2   5.1.3   5.1.3   devDependencies https://github.com/prettier/eslint-plugin-prettier#readme                               
eslint-plugin-solid        0.13.0  0.13.1  0.13.1  devDependencies https://github.com/solidjs-community/eslint-plugin-solid#readme                         
postcss                    8.4.32  8.4.33  8.4.33  devDependencies https://postcss.org/                                                                    
prettier                   3.1.1   3.2.4   3.2.4   devDependencies https://prettier.io                                                                     
solid-js                   1.8.7   1.8.11  1.8.11  dependencies    https://solidjs.com                                                                     
stylelint                  15.11.0 15.11.0 16.2.0  devDependencies https://stylelint.io                                                                    
stylelint-config-standard  34.0.0  34.0.0  36.0.0  devDependencies https://github.com/stylelint/stylelint-config-standard#readme                           
tailwindcss                3.4.0   3.4.1   3.4.1   devDependencies https://tailwindcss.com                                                                 
vite                       5.0.10  5.0.11  5.0.11  devDependencies https://vitejs.dev                                                                      
vite-plugin-solid          2.8.0   2.8.2   2.8.2   devDependencies https://github.com/solidjs/vite-plugin-solid#readme          
```
edit: and configure renovate or dependabot

File: README.md (hunk 3 above)
Comment (hairmare) [reply README.md]: tbh the whole symlinkery feels a tad off, but this should work

**Comment by c0rydoras** [comment]
@hairmare i updated the deps and configured dependabot

**Review by hairmare — APPROVED** [review]
:shipit:

**Pull request merged**